test/hotspot/jtreg/compiler/vectorapi/VectorSaturatedOperationsTest.java

```diff
@@ -0,0 +1,474 @@
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+* @test
+* @bug 8342677
+* @summary Add IR validation tests for newly added saturated vector add / sub operations
+* @modules jdk.incubator.vector
+* @library /test/lib /
+* @run driver compiler.vectorapi.VectorSaturatedOperationsTest
+*/
+
+package compiler.vectorapi;
+
+import jdk.incubator.vector.*;
+import compiler.lib.ir_framework.*;
+import java.util.stream.IntStream;
+
+public class VectorSaturatedOperationsTest {
+    private static final int COUNT = 2048;
+    private static final VectorSpecies<Long> lspec    = LongVector.SPECIES_PREFERRED;
+    private static final VectorSpecies<Integer> ispec = IntVector.SPECIES_PREFERRED;
+    private static final VectorSpecies<Short> sspec   = ShortVector.SPECIES_PREFERRED;
+    private static final VectorSpecies<Byte> bspec    = ByteVector.SPECIES_PREFERRED;
+
+    private long[]  long_in1;
+    private int[]   int_in1;
+    private short[] short_in1;
+    private byte[]  byte_in1;
+
+    private long[]  long_in2;
+    private int[]   int_in2;
+    private short[] short_in2;
+    private byte[]  byte_in2;
+
+    private long[]  long_out;
+    private int[]   int_out;
+    private short[] short_out;
+    private byte[]  byte_out;
+
+    public static void main(String[] args) {
+        TestFramework testFramework = new TestFramework();
+        testFramework.setDefaultWarmup(5000)
+                     .addFlags("--add-modules=jdk.incubator.vector")
+                     .start();
+    }
+
+    public VectorSaturatedOperationsTest() {
+        byte_in1  = new byte[COUNT];
+        short_in1 = new short[COUNT];
+        int_in1   = new int[COUNT];
+        long_in1  = new long[COUNT];
+
+        byte_in2  = new byte[COUNT];
+        short_in2 = new short[COUNT];
+        int_in2   = new int[COUNT];
+        long_in2  = new long[COUNT];
+        IntStream.range(0, COUNT).forEach(
+            i -> {
+                if ((i & 1) == 0) {
+                    long_in1[i] = Long.MAX_VALUE;
+                    long_in2[i] = i;
+                    int_in1[i]  = Integer.MAX_VALUE;
+                    int_in2[i]  = i;
+                    short_in1[i] = Short.MAX_VALUE;
+                    short_in2[i] = (short)i;
+                    byte_in1[i]  = Byte.MAX_VALUE;
+                    byte_in2[i]  = (byte)i;
+                } else {
+                    long_in1[i] = Long.MIN_VALUE;
+                    long_in2[i] = -i;
+                    int_in1[i]  = Integer.MIN_VALUE;
+                    int_in2[i]  = -i;
+                    short_in1[i] = Short.MIN_VALUE;
+                    short_in2[i] = (short)-i;
+                    byte_in1[i]  = Byte.MIN_VALUE;
+                    byte_in2[i]  = (byte)-i;
```

Comment:
Are these values sufficient?

Example with `SADD_VL`:
if the itt elements are `max_value` and `min_value` -> no overflow -> `-1`
if the ith elements are `-i` and `i` -> no overflow -> `0`

So it seems we are actually not even testing the saturation here, am I correct?

Reply:
I am now randomizing input, and explicitly setting delimiting test inputs.